R/pkg/R/DataFrame.R

```diff
@@ -194,7 +195,13 @@ setMethod("isLocal",
 setMethod("showDF",
           signature(x = "SparkDataFrame"),
           function(x, numRows = 20, truncate = TRUE) {
-            s <- callJMethod(x@sdf, "showString", numToInt(numRows), truncate)
+            if (is.logical(truncate) && truncate) {
+                s <- callJMethod(x@sdf, "showString", numToInt(numRows), numToInt(20))
+            } else if (is.numeric(truncate) && truncate > 0) {
+                s <- callJMethod(x@sdf, "showString", numToInt(numRows), numToInt(truncate))
+            } else {
+                s <- callJMethod(x@sdf, "showString", numToInt(numRows), numToInt(0))
+            }
```

Comment:
I could have used, `as.numeric(truncate)` and then no if checks. But, to retain older behavior we need to check if it is True and set it to 20.